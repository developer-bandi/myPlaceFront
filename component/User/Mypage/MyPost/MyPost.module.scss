.infoBlock {
  display: flex;
  color: #5f5f5f;
  align-items: center;
  height: 40px;

  border-top: 2px solid #c0c0c0;
  border-bottom: 2px solid #c0c0c0;
  padding-left: 20px;
  @media screen and (max-width: 1201px) {
    width: 90%;
    margin: auto;
  }
}

.postBlock {
  display: flex;
  height: 40px;
  align-items: center;
  border-bottom: 1px solid #dddddd;
  padding-left: 20px;
  &:hover {
    background-color: #eaeaea;
  }
  @media screen and (max-width: 1201px) {
    width: 90%;
    margin: auto;
  }
}

.index {
  width: 80px;
}

.title {
  width: 500px;
}

.comment {
  width: 100px;
}

.like {
  width: 100px;
}

.view {
  width: 100px;
}

.date {
  width: 100px;
}
