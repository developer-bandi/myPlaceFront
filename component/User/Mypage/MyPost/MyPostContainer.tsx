import { useIsLabtopOrTabletOrMobile } from "../../../../hooks/mediaQuery";
import useMypage from "../../../../hooks/mypage";
import PostList from "./MyPost";
import useMyPost from "./MyPostHook";

export interface postListcontent {
  id: number;
  title: string;
  content: string;
  nickname: string;
  createdAt: string;
  viewCount: number;
  postlikecount: number;
  comment: number;
}

export interface postListState {
  content?: { count: number; rows: postListcontent[] };
  loading: boolean;
  error: boolean;
}

const MyPostContainer = () => {
  const isLabtopOrTabletOrMobile = useIsLabtopOrTabletOrMobile();
  const { movePostDetailPage } = useMyPost();
  const { serverData, changePage, page } = useMypage("post");
  return (
    <PostList
      postListState={serverData as postListState}
      movePostDetailPage={movePostDetailPage}
      page={page}
      changePage={changePage}
      isLabtopOrTabletOrMobile={isLabtopOrTabletOrMobile}
    />
  );
};

export default MyPostContainer;
