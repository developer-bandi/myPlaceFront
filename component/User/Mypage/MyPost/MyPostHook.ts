import {useRouter} from "next/router";
import {useEffect, useState} from "react";
import {axiosGetMyPost} from "../../../../lib/commonFn/api";

export interface postListcontent {
  id: number;
  title: string;
  content: string;
  nickname: string;
  createdAt: string;
  viewCount: number;
  postlikecount: number;
  comment: number;
}

export interface postListState {
  content?: {count: number; rows: postListcontent[]};
  loading: boolean;
  error: boolean;
}

const useMyPost = () => {
  const router = useRouter();
  const movePostDetailPage = (id: number) => {
    router.push(`/community/postdetail/${id}`);
  };

  return {movePostDetailPage};
};

export default useMyPost;
