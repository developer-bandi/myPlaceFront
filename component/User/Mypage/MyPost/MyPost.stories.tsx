import { action } from "@storybook/addon-actions";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { loader } from "../../../../lib/loader";
import { useIsLabtopOrTabletOrMobile } from "../../../../hooks/mediaQuery";
import MyPost from "./MyPost";

export default {
  title: "User/Mypage/MyPost",
  component: MyPost,
} as ComponentMeta<typeof MyPost>;

const Template: ComponentStory<typeof MyPost> = (arg) => {
  const isLabtopOrTabletOrMobile = useIsLabtopOrTabletOrMobile();

  return (
    <MyPost {...arg} isLabtopOrTabletOrMobile={isLabtopOrTabletOrMobile} />
  );
};

const makeUrl = loader({ width: 600, height: 400 });

const nomarlContent = {
  count: 10,
  rows: new Array(10).fill(0).map((data, index) => {
    return {
      id: index + 1,
      title: "테스트 제목입니다",
      content: "안녕하세요 테스트 내용입니다",
      nickname: "place",
      viewCount: 11,
      createdAt: "2022-11-11",
      postlikecount: 1,
      comment: 0,
    };
  }),
};

export const isLoading = Template.bind({});
isLoading.args = {
  postListState: { loading: true, error: false },
};

export const isError = Template.bind({});
isError.args = {
  postListState: { loading: false, error: true },
};

export const isNoResult = Template.bind({});
isNoResult.args = {
  postListState: {
    content: { count: 0, rows: [] },
    loading: false,
    error: false,
  },
};

export const isNomarl = Template.bind({});
isNomarl.args = {
  postListState: { content: nomarlContent, loading: false, error: false },
  page: 1,
  movePostDetailPage: action("movePostDetailPage"),
  changePage: () =>
    new Promise((resolve) => {
      action("changePage")();
      resolve();
    }),
};
