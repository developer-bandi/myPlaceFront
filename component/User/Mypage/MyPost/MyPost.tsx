import Image from "next/image";
import MyPageNavigation from "../Common/navigation/MyPageNavigation";
import searchResultLoading from "../../../../public/searchResultLoading.gif";
import styles from "./MyPost.module.scss";
import mypage from "../../../../lib/styles/mypage.module.scss";
import {setDateYearMonthDay} from "../../../../lib/commonFn/date";
import {postListState} from "./MyPostHook";
import {GrView} from "react-icons/gr";
import {FaRegComment} from "react-icons/fa";
import {BiLike} from "react-icons/bi";
import PageNation from "../../../Common/PageNation/PageNation";

interface PostListProps {
  postListState: postListState;
  movePostDetailPage: (id: number) => void;
  page: number;
  changePage: (page: number) => Promise<void>;
  isLabtopOrTabletOrMobile: boolean;
}

const MyPost = ({
  postListState,
  movePostDetailPage,
  page,
  changePage,
  isLabtopOrTabletOrMobile,
}: PostListProps) => {
  if (postListState.loading) {
    return (
      <div className={mypage.mainBlock} data-testid="loading">
        {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
        <div className={mypage.subBlock}>
          <h1 className={mypage.title}>작성 포스트</h1>
          <div className={mypage.loading}>
            <Image src={searchResultLoading} alt="loading"></Image>
          </div>
        </div>
      </div>
    );
  } else {
    if (postListState.content === undefined && postListState.error !== null) {
      return (
        <div className={mypage.mainBlock} data-testid="error">
          {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
          <div className={mypage.subBlock}>
            <h1 className={mypage.title}>작성 포스트</h1>
            <div className={mypage.error}>에러발생</div>
          </div>
        </div>
      );
    } else if (postListState.content?.count === 0) {
      return (
        <div className={mypage.mainBlock} data-testid="noResult">
          {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
          <div className={mypage.subBlock}>
            <h1 className={mypage.title}>작성 포스트</h1>
            <div className={mypage.noResult}>포스트 없음</div>
          </div>
        </div>
      );
    } else {
      return (
        <div className={mypage.mainBlock} data-testid="result">
          {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
          <div className={mypage.subBlock}>
            <h1 className={mypage.title}>작성 포스트</h1>
            {postListState.content?.rows.map((post, index) => {
              return (
                <div
                  className={styles.postBlock}
                  onClick={() => movePostDetailPage(post.id)}
                  key={post.id}
                >
                  <div className={styles.topBlock}>
                    <div className={styles.user}>{post.nickname}</div>
                    <div className={styles.date}>
                      {setDateYearMonthDay(post.createdAt)}
                    </div>
                  </div>
                  <div className={styles.title}>{post.title}</div>
                  <p className={styles.content}>{post.content}</p>
                  <div className={styles.bottomBlock}>
                    <FaRegComment size="20" className={styles.bottomIcon} />
                    <div className={styles.bottomInfo}>{post.comment}</div>
                    <BiLike size="20" className={styles.bottomIcon} />
                    <div className={styles.bottomInfo}>
                      {post.postlikecount}
                    </div>
                    <GrView size="20" className={styles.bottomIcon} />
                    <div className={styles.bottomInfo}>{post.viewCount}</div>
                  </div>
                </div>
              );
            })}
            <PageNation
              page={page}
              changePage={changePage}
              totalCount={postListState.content?.count as number}
              addStyle={"margin"}
              unit={20}
            />
          </div>
        </div>
      );
    }
  }
};

export default MyPost;
