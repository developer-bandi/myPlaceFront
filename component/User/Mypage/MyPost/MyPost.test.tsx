import {act, render, renderHook, screen, waitFor} from "@testing-library/react";
import axios from "axios";
import {useRouter} from "next/router";
import MyPost from "./MyPost";
import useMyPost from "./MyPostHook";

jest.mock("axios");
jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));
const routerMock = {
  pathname: "",
  push: jest.fn(),
};
(useRouter as jest.Mock).mockReturnValue(routerMock);
const axiosMock = axios as jest.Mocked<typeof axios>;
const postListcontentMock = {
  count: 1,
  rows: [
    {
      id: 1,
      title: "testTitle",
      content: "testContent",
      nickname: "testNickname",
      viewCount: 1,
      createdAt: "testCreatedAt",
      postlikecount: 1,
      comment: 1,
    },
  ],
};

describe("myComment Presentational test", () => {
  describe("데스크톱 일 경우", () => {
    it("로딩중", () => {
      const utils = render(
        <MyPost
          postListState={{
            loading: true,
            error: false,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={false}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("loading");
    });
    it("에러발생", () => {
      const utils = render(
        <MyPost
          postListState={{
            loading: false,
            error: true,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={false}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("error");
    });
    it("결과 없음", () => {
      const utils = render(
        <MyPost
          postListState={{
            content: {count: 0, rows: []},
            loading: false,
            error: false,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={false}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("noResult");
    });
    it("정상 출력", () => {
      const utils = render(
        <MyPost
          postListState={{
            content: postListcontentMock,
            loading: false,
            error: false,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={false}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("result");
    });
  });
  describe("노트북 사이즈 이하일 경우", () => {
    it("로딩중", () => {
      const utils = render(
        <MyPost
          postListState={{
            loading: true,
            error: false,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={true}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("loading");
    });
    it("에러발생", () => {
      const utils = render(
        <MyPost
          postListState={{
            loading: false,
            error: true,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={true}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("error");
    });
    it("결과 없음", () => {
      const utils = render(
        <MyPost
          postListState={{
            content: {count: 0, rows: []},
            loading: false,
            error: false,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={true}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("noResult");
    });
    it("정상 출력", () => {
      const utils = render(
        <MyPost
          postListState={{
            content: postListcontentMock,
            loading: false,
            error: false,
          }}
          movePostDetailPage={jest.fn()}
          page={1}
          changePage={jest.fn()}
          isLabtopOrTabletOrMobile={true}
        />
      );
      expect(utils.container).toMatchSnapshot();
      screen.getByTestId("result");
    });
  });
});

describe("MyPost Hook 테스트", () => {
  it("movePostDetailPage 함수 테스트", () => {
    const {result} = renderHook(() => useMyPost());
    act(() => {
      result.current.movePostDetailPage(1);
    });
    expect(routerMock.push).toBeCalledWith("/community/postdetail/1");
  });
});
