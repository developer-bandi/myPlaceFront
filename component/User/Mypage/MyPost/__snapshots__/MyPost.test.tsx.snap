// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`myComment Presentational test 노트북 사이즈 이하일 경우 결과 없음 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="noResult"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="noResult"
      >
        포스트 없음
      </div>
    </div>
  </div>
</div>
`;

exports[`myComment Presentational test 노트북 사이즈 이하일 경우 로딩중 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="loading"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="loading"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="loading"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`myComment Presentational test 노트북 사이즈 이하일 경우 에러발생 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="error"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="error"
      >
        에러발생
      </div>
    </div>
  </div>
</div>
`;

exports[`myComment Presentational test 노트북 사이즈 이하일 경우 정상 출력 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="result"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="postBlock"
      >
        <div
          class="topBlock"
        >
          <div
            class="user"
          >
            testNickname
          </div>
          <div
            class="date"
          >
            testCreatedAt
          </div>
        </div>
        <div
          class="title"
        >
          testTitle
        </div>
        <p
          class="content"
        >
          testContent
        </p>
        <div
          class="bottomBlock"
        >
          <svg
            class="bottomIcon"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32C114.6 32 0 125.1 0 240c0 47.6 19.9 91.2 52.9 126.3C38 405.7 7 439.1 6.5 439.5c-6.6 7-8.4 17.2-4.6 26S14.4 480 24 480c61.5 0 110-25.7 139.1-46.3C192 442.8 223.2 448 256 448c141.4 0 256-93.1 256-208S397.4 32 256 32zm0 368c-26.7 0-53.1-4.1-78.4-12.1l-22.7-7.2-19.5 13.8c-14.3 10.1-33.9 21.4-57.5 29 7.3-12.1 14.4-25.7 19.9-40.2l10.6-28.1-20.6-21.8C69.7 314.1 48 282.2 48 240c0-88.2 93.3-160 208-160s208 71.8 208 160-93.3 160-208 160z"
            />
          </svg>
          <div
            class="bottomInfo"
          >
            1
          </div>
          <svg
            class="bottomIcon"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 8h-5.612l1.123-3.367c.202-.608.1-1.282-.275-1.802S14.253 2 13.612 2H12c-.297 0-.578.132-.769.36L6.531 8H4c-1.103 0-2 .897-2 2v9c0 1.103.897 2 2 2h13.307a2.01 2.01 0 0 0 1.873-1.298l2.757-7.351A1 1 0 0 0 22 12v-2c0-1.103-.897-2-2-2zM4 10h2v9H4v-9zm16 1.819L17.307 19H8V9.362L12.468 4h1.146l-1.562 4.683A.998.998 0 0 0 13 10h7v1.819z"
            />
          </svg>
          <div
            class="bottomInfo"
          >
            1
          </div>
          <svg
            class="bottomIcon"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12,21 C7,21 1,16 1,12 C1,8 7,3 12,3 C17,3 23,8 23,12 C23,16 17,21 12,21 Z M12,7 C9.23875,7 7,9.23875 7,12 C7,14.76125 9.23875,17 12,17 C14.76125,17 17,14.76125 17,12 C17,9.23875 14.76125,7 12,7 L12,7 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
          <div
            class="bottomInfo"
          >
            1
          </div>
        </div>
      </div>
      <ul
        class="pageButtonList margin"
      >
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
              transform="matrix(-1 0 0 1 24 0)"
            />
          </svg>
        </li>
        <li
          class="pageButton selected"
        >
          1
        </li>
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`myComment Presentational test 데스크톱 일 경우 결과 없음 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="noResult"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="noResult"
      >
        포스트 없음
      </div>
    </div>
  </div>
</div>
`;

exports[`myComment Presentational test 데스크톱 일 경우 로딩중 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="loading"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="loading"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="loading"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`myComment Presentational test 데스크톱 일 경우 에러발생 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="error"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="error"
      >
        에러발생
      </div>
    </div>
  </div>
</div>
`;

exports[`myComment Presentational test 데스크톱 일 경우 정상 출력 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="result"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 포스트
      </h1>
      <div
        class="postBlock"
      >
        <div
          class="topBlock"
        >
          <div
            class="user"
          >
            testNickname
          </div>
          <div
            class="date"
          >
            testCreatedAt
          </div>
        </div>
        <div
          class="title"
        >
          testTitle
        </div>
        <p
          class="content"
        >
          testContent
        </p>
        <div
          class="bottomBlock"
        >
          <svg
            class="bottomIcon"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32C114.6 32 0 125.1 0 240c0 47.6 19.9 91.2 52.9 126.3C38 405.7 7 439.1 6.5 439.5c-6.6 7-8.4 17.2-4.6 26S14.4 480 24 480c61.5 0 110-25.7 139.1-46.3C192 442.8 223.2 448 256 448c141.4 0 256-93.1 256-208S397.4 32 256 32zm0 368c-26.7 0-53.1-4.1-78.4-12.1l-22.7-7.2-19.5 13.8c-14.3 10.1-33.9 21.4-57.5 29 7.3-12.1 14.4-25.7 19.9-40.2l10.6-28.1-20.6-21.8C69.7 314.1 48 282.2 48 240c0-88.2 93.3-160 208-160s208 71.8 208 160-93.3 160-208 160z"
            />
          </svg>
          <div
            class="bottomInfo"
          >
            1
          </div>
          <svg
            class="bottomIcon"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 8h-5.612l1.123-3.367c.202-.608.1-1.282-.275-1.802S14.253 2 13.612 2H12c-.297 0-.578.132-.769.36L6.531 8H4c-1.103 0-2 .897-2 2v9c0 1.103.897 2 2 2h13.307a2.01 2.01 0 0 0 1.873-1.298l2.757-7.351A1 1 0 0 0 22 12v-2c0-1.103-.897-2-2-2zM4 10h2v9H4v-9zm16 1.819L17.307 19H8V9.362L12.468 4h1.146l-1.562 4.683A.998.998 0 0 0 13 10h7v1.819z"
            />
          </svg>
          <div
            class="bottomInfo"
          >
            1
          </div>
          <svg
            class="bottomIcon"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12,21 C7,21 1,16 1,12 C1,8 7,3 12,3 C17,3 23,8 23,12 C23,16 17,21 12,21 Z M12,7 C9.23875,7 7,9.23875 7,12 C7,14.76125 9.23875,17 12,17 C14.76125,17 17,14.76125 17,12 C17,9.23875 14.76125,7 12,7 L12,7 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
          <div
            class="bottomInfo"
          >
            1
          </div>
        </div>
      </div>
      <ul
        class="pageButtonList margin"
      >
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
              transform="matrix(-1 0 0 1 24 0)"
            />
          </svg>
        </li>
        <li
          class="pageButton selected"
        >
          1
        </li>
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
